sqld/src/query_analysis.rs

```diff
@@ -62,20 +62,34 @@ impl StmtKind {
                 | Stmt::CreateIndex { .. },
             ) => Some(Self::Write),
             Cmd::Stmt(Stmt::Select { .. }) => Some(Self::Read),
-            Cmd::Stmt(Stmt::Pragma(name, body)) if is_pragma_allowed(name, body.as_ref()) => {
-                Some(Self::Write)
+            Cmd::Stmt(Stmt::Pragma(name, body)) => {
+                if is_ro_pragma(name, body.as_ref()) {
+                    Some(Self::Read)
+                } else if is_pragma_allowed(name, body.as_ref()) {
+                    Some(Self::Write)
+                } else {
+                    None
+                }
             }
             _ => None,
         }
     }
 }
 
+fn is_ro_pragma(name: &QualifiedName, _body: Option<&PragmaBody>) -> bool {
+    matches!(name, QualifiedName {
+            db_name: None,
+            name,
+            alias: None,
+        } if name.0.starts_with("index_")|| name.0 == "encoding" || name.0 == "function_list" || name.0 == "module_list" || name.0.starts_with("table_"))
```

Comment:
Some pragmas are only read-only if they don't have parameters, e.g. `encoding`:
```
PRAGMA encoding;
PRAGMA encoding = 'UTF-8';
PRAGMA encoding = 'UTF-16';
PRAGMA encoding = 'UTF-16le';
PRAGMA encoding = 'UTF-16be';
```

I think we can still just merge as is and fix it in a follow-up by inspecting the `body: Option<&PragmaBody>` parameter, but I'll leave the finall call to @MarinPostma

Reply:
or we could whitelist encoding as write. The others are mostly list of stuff, and should be ro.

Reply:
I need to test, but I think it's as simple as: body is Some => write, body is None => read

Reply:
yup, exactly:
pragma encoding = 'utf8':
```rust
name: QualifiedName { db_name: None, name: Name("encoding"), alias: None }, body Some(Equals(Name(Name("utf8"))))
```
pragma encoding:
```rust
name: QualifiedName { db_name: None, name: Name("encoding"), alias: None }, body None
```